### PR TITLE
SIGTERM handling

### PR DESCRIPTION
**Pull request by meliaj**
We have our own SIGTERM handling https://github.com/Clever/sfncli/blob/18864ae18d813dc9a743efb8cec7935f423ba36c/cmd/sfncli/runner.go#L170-L199 which isn't being properly executed because when sfncli receives a SIGTERM we cancel the context which we're passing through to the command we're running.

When the command context is cancelled, we kill the process: https://golang.org/src/os/exec/exec.go?#L404, which is NOT what we want, since our SIGTERM handling should do this for us in a more graceful way.

### CODE REVIEW

**Comment by mohit** [comment]
At what point are we cancelling the context on sigterm? I don’t see a context cancel in the sigterm handling.

**Comment by meliaj** [comment]
Conversation from Slack:
@mohit: As I understand there are two cases here.
1. bug where our sigterm handling does not actually give the child process enough time to exit.
2. On errors or heartbbeat fails we currently expect to kill the child process immediately without wait time.
fixing 1 can be done by removing context from the `cmd.Exec()` but that could lead to cases of heartbeat failure where we might have the same job being run by two containers. (edited) 
the best short term change imo is to disconnect taskCtx from mainCtx (start it using context.Background() ) and manually cancel it after the 25sec wait.

**Comment by meliaj** [comment]
Tested locally that this handles SIGTERM at the sfncli main context level, allows the command signal handling to proceed, and the task is then cancelled/killed.

File: cmd/sfncli/runner.go [hunk 1]
```diff
@@ -187,6 +189,7 @@ func (t *TaskRunner) handleSignals(ctx context.Context) {
 				t.receivedSigterm = true
 				go func(pidtokill int) {
 					time.Sleep(t.sigtermGracePeriod)
+					t.ctxCancel()
```
Comment (mohit) [inline cmd/sfncli/runner.go]: signalProcess before killing the taskRunner?

**Pull request merged**